src/main/java/products/Product.java

```diff
@@ -0,0 +1,33 @@
+package products;
+
+import java.util.List;
+
+public class Product {
+
+    private List<Category> categoryList;
+
+    private String name;
+
+    private int count;
+    private int prise;
+
+    public List<Category> getCategories() {
```

Comment:
Making list final solves the problem of returning it directly. Okay

Reply:
No, it will not.
`final List<Category> categoryList = new ArrayList<>();` 
`final` doesn't allow to reassign link `categoryList` but it allows to add elements to list. e.g.
So caller of your method can do the following, that may cause bugs. 
```
List<Category> categoryList product.getCategories()
categotyList.clear()
```
So I recommend to return copies or immutable copies.

Reply:
Got it. Thank you!